### ISSUE TITLE
Add platform independent way to specify paths

### ISSUE THREAD

**Issue by qiemem** [NetLogo/NetLogo#1061]
Currently, paths are just specified as flat strings, which means that there's no way to specify a path that will work on both unix-like OSs and Windows. I see two ways of solving this. Something similar to Python's `os.path.join`:

```
(path "path" "to" "file.csv")
```

would return `"path/to/file.csv"` on *nix and `"path\to\file.csv"` on Windows. A name other than "path" might be preferable, as discussing paths in ABM often has a very different meaning.

Alternatively, we could have something like Scala's `/` operator:

```
"path" sep "to" sep "file.csv"
```

I think I would prefer the first solution.

**Comment by mrerrormessage** [comment]
Can you specify which methods in particular you are talking about? All things being equal, I would probably prefer see us use `java.net.URI`s in most places where we now use path strings since URI's are a representation which would also allow (at least hypothetically) opening things from the internet. If we're restricting ourselves to the immediate file system, Java 7 introduced the awesome new [java.nio.file.Path](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html) which I would prefer over both flat strings and sequences of strings. Another big plus of using `Path` is that it converts pretty seamlessly to/from `file:///` URI's. I've made extensive use of this in the recent model saving/loading changes.

I will say there is major work to be done in a lot of areas to rip out the string paths entirely. I propose that we start with the api-visible methods first and work our way in towards classes and methods rarely used by extensions or embedding programs (like the models library classes).

**Comment by qiemem** [comment]
I'm sorry, I meant add a primitive to the language that allows people to do this in NetLogo. I'll edit the title.

**Comment by qiemem** [comment]
For a little context: this came up as Arthur and I were discussing specifying LevelSpace child models in a platform independent way.

**Comment by ERSUCC** [comment]
Maybe something has changed since this issue was first opened, but as of at least NetLogo 7.0.3, you can use the forward path separator on all platforms and nothing breaks.

**Comment by TheBizzle** [comment]
Yeah, I'm not able to deduce what prompted this ticket.  `file-open` handles `/` just fine on Windows, even as far back as NetLogo 5.0 (which predates this ticket).  I'm going to close this, but would be alright with reopening it in the future, if someone provides a good use case for where our cross-platform handling of slashes could use improvement.
